### PR TITLE
VZ-7982 Implement the Istio dashboards in Grafana

### PR DESCRIPTION
**Pull request by davisgomes**
Provide a summary of the change and which issue (i.e. ticket) is fixed.
If there are any dependencies, for example on a PR in another repository, list those.

### CODE REVIEW

File: platform-operator/helm_config/charts/verrazzano-grafana-dashboards/templates/configmaps.yaml [hunk 1]
```diff
@@ -10,6 +10,9 @@ items:
   {{- range $path, $_ :=  .Files.Glob  "**/*.json" }}
   {{- $fileName := base $path }}
   {{- $cmName := trimPrefix "dashboards/" $path | trimSuffix ".json" | replace "/" "-" | replace "_" "-" | replace " " "-" | lower }}
+  {{- $subdir := trimSuffix $fileName $path | trimSuffix "/" | base }}
+# If Istio is disabled, skip deploying the Istio ConfigMaps
+{{- if or $.Values.istioEnabled (not (eq $subdir "Istio")) }}
```
Comment (desagar) [inline platform-operator/helm_config/charts/verrazzano-grafana-dashboards/templates/configmaps.yaml:15]: this logic is a bit confusing to me. Why do we need the second condition?

Comment (davisgomes) [reply platform-operator/helm_config/charts/verrazzano-grafana-dashboards/templates/configmaps.yaml:15]: This logic is a boolean simplification of this condition:

File: platform-operator/controllers/verrazzano/component/grafanadashboards/dashboards.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import (
+	"github.com/verrazzano/verrazzano/pkg/bom"
+	"github.com/verrazzano/verrazzano/pkg/vzcr"
+	"github.com/verrazzano/verrazzano/platform-operator/controllers/verrazzano/component/spi"
+)
+
+// AppendOverrides builds the set of Grafana dashboard overrides for the Helm install
+func AppendOverrides(compContext spi.ComponentContext, _ string, _ string, _ string, kvs []bom.KeyValue) ([]bom.KeyValue, error) {
+	if !vzcr.IsIstioEnabled(compContext.EffectiveCR()) || !vzcr.IsIstioEnabled(compContext.EffectiveCRV1Beta1()) {
```
Comment (desagar) [inline platform-operator/controllers/verrazzano/component/grafanadashboards/dashboards.go:14]: can't we use enabled.IsIstioEnabled? Why do we have to pass in both here?

Comment (davisgomes) [reply platform-operator/controllers/verrazzano/component/grafanadashboards/dashboards.go:14]: The IsIstioEnabled function accepts a runtime object. I believe I have to check both of the effective CRs to make sure if it's disabled unless I'm misunderstanding.

**Review by desagar — APPROVED** [review]

**Pull request merged**